OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -1035,9 +1039,22 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == REQUEST_SHOW_COURSE_UNIT_DETAIL && resultCode == Activity.RESULT_OK
                 && data != null) {
-            final CourseComponent outlineComp = courseManager.getComponentByIdFromAppLevelCache(
-                    courseData.getCourseId(), courseComponentId);
-            navigateToCourseUnit(data, courseData, outlineComp);
+            if (data.getBooleanExtra(AppConstants.COURSE_UPGRADED, false)) {
+                courseData.setMode(EnrollmentMode.VERIFIED.toString());
+                if (!isOnCourseOutline) {
+                    Intent resultData = new Intent();
+                    resultData.putExtra(AppConstants.COURSE_UPGRADED, true);
+                    requireActivity().setResult(Activity.RESULT_OK, resultData);
```

Comment:
deserve docstring, cuz we are setting up results on activity context and not using anywhere else.

Reply:
Added